terms/elliptic_curve.md

```diff
@@ -0,0 +1,110 @@
+
+#### Extension Fields
+
+$i = \sqrt{-1}$ 
+
+Basically, extension fields work by taking an existing field, then "inventing" a new element and defining the relationship between that element and existing elements (in this case, $i^2 + 1 = 0$) (Also called [Quadratic field](quadratic_field.md)).
+
+$a / b = (a * b^{p^{2}-2}) \pmod {p}$
+
+$x^{p² - 1} \equiv 1 \pmod p$ => $p^2 - 1$ is the **order of the multiplicative group in the field**
+
+For any function $f(x)$ such that $f(i) = j$
+
+#### Prime Fields vs Real Numbers
+
+With real numbers, the [Fundamental Theorem of Algebra](https://en.wikipedia.org/wiki/Fundamental_theorem_of_algebra) ensures that the quadratic extension that we call the complex numbers is “complete” — you cannot extend it further, because for any mathematical relationship (at least, any mathematical relationship defined by an algebraic formula) that you can come up with between some new element j and the existing complex numbers, it’s possible to come up with at least one complex number that already satisfies that relationship. With prime fields, however, we do not have this issue, and so we can go further and make cubic extensions (where the mathematical relationship between some new element w and existing field elements is a cubic equation, so 1, w and w² are all linearly independent of each other), higher-order extensions, extensions of extensions, etc. And it is these kinds of supercharged modular complex numbers that elliptic curve pairings are built on.
+
+### Elliptic Curve Pairings
+Now, on to elliptic curve pairings. An elliptic curve pairing is a map G2 x G1 -> Gt, where:
+
+- **G1** is an elliptic curve, where points satisfy an equation of the form $y^2 = x^3 + b$, and where both coordinates are elements of $F_p$ (ie. they are simple numbers, except arithmetic is all done modulo some prime number)
+- **G2** is an elliptic curve, where points satisfy the same equation as G1, except where the coordinates are elements of **$F_{p^{12}}$** (ie. they are the supercharged complex numbers we talked about above; we define a new “magic number” w, which is defined by a 12th degree polynomial like $w^{12} - 18 * w^6 + 82 = 0$)
+- **Gt** is the type of object that the result of the elliptic curve goes into. In the curves that we look at, **Gt** is $F_{p^{12}}$ (the same supercharged complex number as used in **G2**)
+
+The main property that it must satisfy is [bilinearity property](bilinear_map.md), which in this context means that:
+
+- e(P, Q + R) = e(P, Q) * e(P, R)
+- e(P + Q, R) = e(P, R) * e(Q, R)
+
+There are two other important criteria:
+
+- **Efficient computability** (eg. we can make an easy pairing by simply taking the discrete logarithms of all points and multiplying them together, but this is as computationally hard as breaking elliptic curve cryptography in the first place, so it doesn’t count)
+- **Non-degeneracy** (sure, you could just define e(P, Q) = 1, but that’s not a particularly useful pairing)
```

Comment:
According to wikipedia, non-degeneracy can be understood as for every input possible, there won't be a "special" case, for example, circle is a degenerated ellipse. Therefore, non-degeneracy property of a pairing function is no P and Q exists that e(P, Q) = 1